core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java

```diff
@@ -1520,7 +1530,31 @@ public long saveProjectOverviewNote(ProjectOverview overview, Note note) throws
   @Override
   public long saveRun(Run run) throws IOException {
     if (runStore != null) {
-      return runStore.save(run);
+      long id;
+      if (run.getId() == AbstractRun.UNSAVED_ID) {
+        runStore.save(run);
+        try {
+          String name = namingScheme.generateNameFor(run);
+          run.setName(name);
+
+          validateNameOrThrow(run, namingScheme);
+          Number newId = (long) runStore.save(run);
+          if (newId.longValue() != run.getId()) {
```

Comment:
These will never be equal because `run.id` was never changed. The check is unnecessary now though. A similar check was used previously because the generated name was based on a predicted ID before the Run was actually saved. Since it is now saved first, the ID is solid.

Above (L1535) should be changed to `run.setId(runStore.save(run));` though